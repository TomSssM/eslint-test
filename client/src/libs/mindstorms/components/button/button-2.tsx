import React from 'react';
import { Button } from './button';

const dd = <Button icon={'dd'} size='dd' view />;

// eslint-disable-next-line no-console
console.log("ðŸš€ ~ file: button-2.tsx ~ line 4 ~ dd", dd);
