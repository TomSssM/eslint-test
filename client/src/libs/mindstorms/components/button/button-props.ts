import React from 'react';

// The size of the button depends on its height
export const enum ButtonSize {
    S56 ='s56',
    S44 = 's44',
    S32 = 's32'
}

export const enum ButtonView {
    Contrast = 'contrast',
    Shade = 'shade'
}

// Auto - width to fit content, Max - width to fit container
export const enum ButtonWidth {
    Auto = 'auto',
    Max = 'max'
}

export const enum ButtonHtmlType {
    Button = 'button',
    Submit = 'submit',
    Reset = 'reset'
}

export interface BaseButtonProps {
    size: ButtonSize;
    view: ButtonView;
    isDisabled?: boolean;
    inProgress?: boolean;
    className?: string;
    htmlType?: ButtonHtmlType;
    onClick?: React.MouseEventHandler;
    onFocus?: React.FocusEventHandler;
    onBlur?: React.FocusEventHandler;
}

export type TextButtonProps = {
    children: string;
    width?: ButtonWidth;
    iconLeft?: React.ReactChild;
    iconRight?: React.ReactChild;
} & BaseButtonProps;

export type IconButtonProps = {
    icon: React.ReactChild;
} & BaseButtonProps;

export type ButtonProps = TextButtonProps | IconButtonProps;
