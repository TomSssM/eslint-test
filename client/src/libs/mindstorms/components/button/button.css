.button {
    border: none;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: transform .1s ease-out, color .15s linear;
    color: var(--clr-el-gray-080);

    &:active {
        transition: none;
        transform: scale(.96);
    }

    &:disabled,
    &:disabled:hover {
        transform: scale(1);
        background-color: var(--clr-el-gray-020);
        color: var(--crl-el-gray-050);
        cursor: default;
    }

    &_size {
        &_s32 {
            border-radius: 10px;

            &.button_mode_text {
                padding: 5px 8px;
            }

            &.button_mode_icon {
                padding: 8px;
            }

            .button__text {
                font-size: 14px;
                line-height: 20px;
                padding-bottom: 2px;
            }
        }

        &_s44 {
            border-radius: 14px;

            &.button_mode_text {
                padding: 10px 14px;
            }

            &.button_mode_icon {
                padding: 10px;
            }

            .button__text {
                font-size: 16px;
                line-height: 24px;
            }
        }

        &_s56 {
            border-radius: 14px;

            &.button_mode_text {
                padding: 13px 14px;
            }

            &.button_mode_icon {
                padding: 16px;
            }

            .button__text {
                font-size: 20px;
                line-height: 28px;
                padding-bottom: 2px;
            }
        }
    }

    &_view {
        &_contrast {
            background-color: var(--clr-bg-gray-1);

            &:hover {
                background-color: var(--clr-bg-gray-2);
            }

            &:active {
                background-color: var(--clr-bg-gray-2);
            }
        }

        &_shade {
            background-color: var(--clr-el-gray-020);

            &:hover {
                background-color: var(--clr-el-gray-030);
            }

            &:active {
                background-color: var(--clr-el-gray-030);
            }
        }
    }

    &_width {
        &_max {
            width: 100%
        }

        &_auto {
            width: auto;
            max-width: max-content;
        }
    }

    &__text {
        font-weight: 400;
        text-align: center;
        white-space: nowrap;
        user-select: none;

        &_with-left-icon {
            margin-left: 8px;
        }

        &_with-right-icon {
            margin-right: 8px;
        }
    }
}
