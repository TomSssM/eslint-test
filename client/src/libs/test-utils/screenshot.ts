import { readFileSync } from 'fs';
import ReactDOMServer from 'react-dom/server';
import React from 'react';
import postcss from 'postcss';
import postcssNested from 'postcss-nested';

export async function makeScreenshot(
    component: React.ReactElement,
    options?: { width: number; height: number },
): Promise<void> {
    const { width = 256, height = 256 } = options ?? {};

    const [html, styles] = await getComponentMarkup(component);

    /*
        Snapshot is needed not to take a screenshot when the snapshot has not changed
        It checked in toMatchScreenshot function
     */
    expect(`${styles}\n${html}`).toMatchSnapshot();

    const markup = `
        <style>${styles}</style>
        <div>
            ${html}
        </div>
    `;

    await expect(markup).toMatchScreenshot({ width, height });
}

async function getComponentMarkup(
    component: React.ReactElement,
): Promise<[string, string]> {
    const html = ReactDOMServer.renderToStaticMarkup(component);
    const styles = await getComponentStyles();
    return [html, styles];
}

async function getComponentStyles(): Promise<string> {
    // All css file paths that are used in the component are written to the global variable using babel visitor
    const paths: string[] = window.includedCssFiles || [];
    let resultStyles = '';
    for (let i = 0; i < paths.length; i++) {
        const css = readFileSync(paths[i]);
        const result = await postcss([postcssNested]).process(css, {
            from: paths[i],
        });
        resultStyles += result.css;
    }
    return resultStyles;
}
