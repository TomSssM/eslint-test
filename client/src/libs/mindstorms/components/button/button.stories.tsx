import React from 'react';
import { PlusIcon } from '../icon/PlusIcon';
import { Button } from './button';
import {
    BaseButtonProps,
    ButtonProps,
    ButtonSize,
    ButtonView,
    IconButtonProps,
    TextButtonProps,
} from './button-props';
import {
    getCommonOptions,
    getPrimaryStory,
    getStoryForProp,
} from '../../storybook';

const FIGMA_URL =
    'https://www.figma.com/file/i6xFWsF5ha8oRgMHUjnkGQ/New-Metrica.-Library?node-id=1310%3A1353';
const defaultProps: BaseButtonProps = {
    view: ButtonView.Shade,
    size: ButtonSize.S32,
    isDisabled: false,
} as const;

const defaultTextProps: TextButtonProps = {
    ...defaultProps,
    children: 'Button',
    iconLeft: <PlusIcon fill="currentColor" size={16} />,
} as const;

const dd = 12;

dd;

const { title, component, decorators, parameters } = getCommonOptions({
    component: Button,
    figma: FIGMA_URL,
    description: 'Button',
});

export default {
    title,
    component,
    decorators,
    argTypes: {
        onClick: { action: 'click' },
    },
    parameters,
};

export const Main = getPrimaryStory<ButtonProps>({
    Component: Button,
    defaultProps: defaultTextProps,
});

export const View = getStoryForProp<ButtonProps, 'view'>({
    Component: Button,
    defaultProps: defaultTextProps,
    properties: [
        { name: 'view', values: [ButtonView.Shade, ButtonView.Contrast] },
    ],
});

export const Size = getStoryForProp<ButtonProps, 'size'>({
    Component: Button,
    defaultProps: defaultTextProps,
    properties: [
        {
            name: 'size',
            values: [ButtonSize.S32, ButtonSize.S44, ButtonSize.S56],
        },
    ],
});

export const Disabled = getStoryForProp<ButtonProps, 'isDisabled'>({
    Component: Button,
    defaultProps: defaultTextProps,
    properties: [{ name: 'isDisabled', values: [true] }],
});

export const IconButton = getStoryForProp<IconButtonProps, 'icon'>({
    Component: Button,
    defaultProps: {
        ...defaultProps,
        icon: <PlusIcon fill="currentColor" size={16} />,
    },
    properties: [
        { name: 'icon', values: [<PlusIcon fill="currentColor" size={16} />] },
    ],
});

export const TextButtonWithIcon = getStoryForProp<
    TextButtonProps,
    'iconLeft' | 'iconRight'
>({
    Component: Button,
    defaultProps: defaultTextProps,
    properties: [
        {
            name: 'iconLeft',
            values: [<PlusIcon fill="currentColor" size={16} />],
        },
        {
            name: 'iconRight',
            values: [<PlusIcon fill="currentColor" size={16} />],
        },
    ],
});

// One story with all configurations for screenshot test
// TODO support IconButtonProps
export const All = getStoryForProp<TextButtonProps, keyof TextButtonProps>({
    Component: Button,
    defaultProps: defaultTextProps,
    properties: [
        {
            name: 'size',
            values: [ButtonSize.S32, ButtonSize.S44, ButtonSize.S56],
        },
        { name: 'view', values: [ButtonView.Shade, ButtonView.Contrast] },
        { name: 'isDisabled', values: [false, true] },
        {
            name: 'iconLeft',
            values: [<PlusIcon fill="currentColor" size={16} />],
        },
        {
            name: 'iconRight',
            values: [<PlusIcon fill="currentColor" size={16} />],
        },
    ],
});
