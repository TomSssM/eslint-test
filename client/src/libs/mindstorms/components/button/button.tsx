import React, { FC, ForwardedRef, forwardRef } from 'react';
import './button.css';
import { bem } from '../../../bem/bem';
import { ButtonProps, ButtonWidth, TextButtonProps } from './button-props';

const cls = bem('button');

const TextButtonContent: FC<TextButtonProps> = (props: TextButtonProps) => {
    const { iconLeft, iconRight } = props;

    return (
        <>
            {iconLeft}
            <span
                className={cls({
                    elem: 'text',
                    mods: {
                        'with-left-icon': !!iconLeft,
                        'with-right-icon': !!iconRight
                    }
                })}
            >
                {props.children}
            </span>
            {iconRight}
        </>
    );
};

export const Button = forwardRef<HTMLButtonElement, ButtonProps>((
    props: ButtonProps,
    ref: ForwardedRef<HTMLButtonElement>,
) => {
    const {
        size,
        view,
    } = props;

    return (
        <button
            ref={ref}
            className={cls({
                mods: {
                    width: 'width' in props ? props.width : ButtonWidth.Auto,
                    size,
                    view,
                    mode: 'icon' in props ? 'icon' : 'text'
                },
                mix: props.className
            })}
            disabled={props.isDisabled}
            type={props.htmlType}
            onClick={props.onClick}
            onFocus={props.onFocus}
            onBlur={props.onBlur}
        >
            {'icon' in props ? props.icon : <TextButtonContent {...props} />}
        </button>
    );
});
