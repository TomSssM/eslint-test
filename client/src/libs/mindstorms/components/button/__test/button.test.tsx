import * as React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Button } from '../button';
import { ButtonProps, ButtonSize, ButtonView } from '../button-props';

import { All } from '../button.stories';
import { makeScreenshot } from '../../../../test-utils/screenshot';

const title = 'button';
const onButtonClick = jest.fn();
const onFocus = jest.fn();
const onBlur = jest.fn();

function renderButton(props?: Partial<ButtonProps>): void {
    render(
        <Button
            view={ButtonView.Shade}
            size={ButtonSize.S32}
            onClick={onButtonClick}
            onFocus={onFocus}
            onBlur={onBlur}
            {...props}
        >
            {title}
        </Button>
    );
}

function emulateClick(): void {
    userEvent.click(screen.getByText(title));
}

describe('Button', () => {
    it('screenshot', async () => {
        await makeScreenshot(<All/>, { width: 400, height: 400 });
    });

    describe('logic', () => {

        beforeEach(() => {
            onButtonClick.mockClear();
            onFocus.mockClear();
            onBlur.mockClear();
        });

        it('should call onClick', () => {
            renderButton();
            emulateClick();

            expect(onButtonClick).toBeCalled();
        });

        it('should not call onClick when button is disabled', () => {
            renderButton({ isDisabled: true });
            emulateClick();

            expect(onButtonClick).not.toBeCalled();
        });

        it('should call onFocus', () => {
            renderButton();
            fireEvent.focus(screen.getByText(title));

            expect(onFocus).toBeCalled();
        });

        it('should call onBlur', () => {
            renderButton();
            const btn = screen.getByText(title);
            fireEvent.focus(btn);
            fireEvent.blur(btn);

            expect(onBlur).toBeCalled();
        });
    });
});
