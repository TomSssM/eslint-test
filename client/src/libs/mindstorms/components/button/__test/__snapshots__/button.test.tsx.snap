// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button screenshot 1`] = `
".button {
    border: none;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: transform .1s ease-out, color .15s linear;
    color: var(--clr-el-gray-080);
}

    .button:active {
        transition: none;
        transform: scale(.96);
    }

    .button:disabled,
    .button:disabled:hover {
        transform: scale(1);
        background-color: var(--clr-el-gray-020);
        color: var(--crl-el-gray-050);
        cursor: default;
    }

    .button_size_s32 {
            border-radius: 10px;
        }

    .button_size_s32.button_mode_text {
                padding: 5px 8px;
            }

    .button_size_s32.button_mode_icon {
                padding: 8px;
            }

    .button_size_s32 .button__text {
                font-size: 14px;
                line-height: 20px;
                padding-bottom: 2px;
            }

    .button_size_s44 {
            border-radius: 14px;
        }

    .button_size_s44.button_mode_text {
                padding: 10px 14px;
            }

    .button_size_s44.button_mode_icon {
                padding: 10px;
            }

    .button_size_s44 .button__text {
                font-size: 16px;
                line-height: 24px;
            }

    .button_size_s56 {
            border-radius: 14px;
        }

    .button_size_s56.button_mode_text {
                padding: 13px 14px;
            }

    .button_size_s56.button_mode_icon {
                padding: 16px;
            }

    .button_size_s56 .button__text {
                font-size: 20px;
                line-height: 28px;
                padding-bottom: 2px;
            }

    .button_view_contrast {
            background-color: var(--clr-bg-gray-1);
        }

    .button_view_contrast:hover {
                background-color: var(--clr-bg-gray-2);
            }

    .button_view_contrast:active {
                background-color: var(--clr-bg-gray-2);
            }

    .button_view_shade {
            background-color: var(--clr-el-gray-020);
        }

    .button_view_shade:hover {
                background-color: var(--clr-el-gray-030);
            }

    .button_view_shade:active {
                background-color: var(--clr-el-gray-030);
            }

    .button_width_max {
            width: 100%
        }

    .button_width_auto {
            width: auto;
            max-width: max-content;
        }

    .button__text {
        font-weight: 400;
        text-align: center;
        white-space: nowrap;
        user-select: none;
    }

    .button__text_with-left-icon {
            margin-left: 8px;
        }

    .button__text_with-right-icon {
            margin-right: 8px;
        }
:root {
    --clr-el-gray-020: #B0BDD633;
    --clr-el-gray-030: #A5B1CA4D;

    --clr-bg-gray-1: #FFFFFF;
    --clr-bg-gray-2: #F6F8FB;

    --clr-el-gray-080: #2F3747;
    --crl-el-gray-050: #475A8080;
}

<div style=\\"margin-bottom:8px\\"><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">size: s32</span></button></div><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s44 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">size: s44</span></button></div><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s56 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">size: s56</span></button></div></div><div style=\\"margin-bottom:8px\\"><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">view: shade</span></button></div><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_contrast button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">view: contrast</span></button></div></div><div style=\\"margin-bottom:8px\\"><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">isDisabled</span></button></div><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_shade button_mode_text\\" disabled=\\"\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">isDisabled</span></button></div></div><div style=\\"margin-bottom:8px\\"><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon\\">iconLeft</span></button></div></div><div style=\\"margin-bottom:8px\\"><div style=\\"margin-right:8px;display:inline-block\\"><button class=\\"button button_width_auto button_size_s32 button_view_shade button_mode_text\\"><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg><span class=\\"button__text button__text_with-left-icon button__text_with-right-icon\\">iconRight</span><svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\"><path color=\\"currentColor\\" fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M12 3C11.4477 3 11 3.44772 11 4V11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H11V20C11 20.5523 11.4477 21 12 21C12.5523 21 13 20.5523 13 20V13H20C20.5523 13 21 12.5523 21 12C21 11.4477 20.5523 11 20 11H13V4C13 3.44772 12.5523 3 12 3Z\\"></path></svg></button></div></div>"
`;
